### PR TITLE
Prevent fetching of oci timing out snap install

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following wraps the installing of the oci image in a timeout. As we
don't actually require the image to be installed when we bootstrap we
can be a bit fast and loose about it. If you don't have timeout
installed (BSD and macosx) then we exit and install the image using
manual intervention/bootstrap.

The code ensures that if it takes longer than we expect we just cancel
it, but importantly ignore the timeout error and return true. That way
we can still install the snap.

I've used 2 minutes because it can take a long time to install a snap on
slower systems, so we don't want to gobble that time up installing an
image that might not install anyway.

## QA steps

Ensure you have microk8s installed.
If you've built the snap before then it's wise to do `lxc delete snapcraft-juju --force`

```sh
snap install snapcraft --classic
snapcraft --use-lxd
sudo snap install *.snap --dangerous --classic
sudo snap logs juju
```

Ensure that we did in fact attempt to pull the images down.

### CODE REVIEW

File: snap/local/wrappers/fetch-oci [hunk 1]
```diff
@@ -8,6 +8,12 @@ if ! which microk8s.kubectl; then
     exit 0
 fi
 
+if ! which timeout; then
+    echo "timeout is not installed."
+    exit 0
+fi
+
+timeout 2m sh <<EOT || echo true
```
Comment (petevg) [inline snap/local/wrappers/fetch-oci:16]: `|| true` should be sufficient here, correct? Or echo "microk8s docker images downloading in bg" if you want to give a user some information.

**Review by petevg — APPROVED** [review]
This makes sense. Haven't had time to QA, and had a minor nitpick w/ the code. But this is a nice, elegant fix. :-)

File: snap/local/wrappers/fetch-oci (hunk 1 above)
Comment (SimonRichardson) [reply snap/local/wrappers/fetch-oci:16]: I think you're right, let me change this before it lands. It would be much better to add a message here.

**Review by manadart — APPROVED** [review]

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**